app/src/main/java/org/mozilla/fenix/gleanplumb/MessageNotificationWorker.kt

```diff
@@ -123,3 +161,32 @@ class MessageNotificationWorker(
         }
     }
 }
+
+/**
+ * When a [Message] [Notification] is dismissed by the user, record telemetry data and update the
+ * dismissed [Message] as 'displayed' so we don't show it again.
+ *
+ * This [Service] is only intended to be used by the [MessageNotificationWorker].
+ */
+class NotificationDismissedService : Service() {
+
+    /**
+     * This service cannot be bound to.
+     */
+    override fun onBind(intent: Intent?): IBinder? = null
+
+    @OptIn(DelicateCoroutinesApi::class) // GlobalScope usage
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        GlobalScope.launch {
+            val messagingStorage = applicationContext.components.analytics.messagingStorage
+            val nimbusMessagingController = NimbusMessagingController(messagingStorage)
+
+            val messageMetadata =
+                JSONObject(intent?.getStringExtra(MESSAGE_METADATA)!!).toMetadata()
+
+            nimbusMessagingController.onMessageDismissed(messageMetadata)
+        }
+
+        return START_REDELIVER_INTENT
+    }
+}
```

Comment:
I love this use of a service. 

I think I'd like to see this (or similar service) catch the `onMessageClicked`, process the action to make a new Intent, and then forward on.

Reply:
Nice suggestion 👍